hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -223,21 +240,25 @@ object HoodieSparkSqlWriter {
               operation.equals(WriteOperationType.UPSERT) ||
               parameters.getOrElse(HoodieWriteConfig.COMBINE_BEFORE_INSERT_PROP.key(),
                 HoodieWriteConfig.COMBINE_BEFORE_INSERT_PROP.defaultValue()).toBoolean
+            val partitionKey = hoodieConfig.getString(DataSourceWriteOptions.PARTITIONPATH_FIELD)
```

Comment:
hi codope! ask a question,  user can drop partition data through the pr ?

Reply:
@xuzifu666 
We have tried Hudi with the PR and it solved the problem with the BigQuery index conflict.

Reply:
@xuzifu666 Yes, you can drop partition fields with this patch.
@Ales-Semchenkov Good to hear!